MixItUp.Desktop/Services/DesktopTelemetryService.cs

```diff
@@ -54,5 +56,18 @@ public void End()
             this.telemetryClient.Flush();
             Task.Delay(1000); // Allow time to flush
         }
+
+        private void RefreshUserDetails()
+        {
+            if (string.IsNullOrEmpty(this.telemetryClient.Context.User.Id))
+            {
+                if (string.IsNullOrEmpty(ChannelSession.Settings.TelemetryUserId))
+                {
+                    ChannelSession.Settings.TelemetryUserId = Guid.NewGuid().ToString();
+                }
+
+                this.telemetryClient.Context.User.Id = ChannelSession.Settings.TelemetryUserId;
```

Comment:
Any reason this isn't just being set when you make the telemetry client?

Reply:
We don't have settings loaded yet? Or a user?

Reply:
If we didn't have settings loaded yet, then you wouldn't be able to set a value in ChannelSession.Settings